src/js/libcs/Accessors.js

```diff
@@ -168,6 +168,22 @@ Accessor.setField = function(geo, field, value) {
         movepointscr(geo, List.turnIntoCSList([value.value[0], value.value[1], CSNumber.real(1)]), "homog");
     }
 
+    if (field === "x" && geo.kind === "P" && geo.movable && value.ctype === "number") {
+        if (CSNumber._helper.isAlmostZero(geo.homog.value[2])) {
+            console.log("cannot set x component without y component of far points");
+        } else {
+            movepointscr(geo, List.turnIntoCSList([value, CSNumber.div(geo.homog.value[1], geo.homog.value[2]), CSNumber.real(1)]), "homog");
+        }
+    }
```

Comment:
But in the original code you'd assign `[x, NaN, 1]` which is far from being better. `[0,0,0]` is `NaHC` (not the chemical formula, but short for `Not a Homogeneous Coordinate`)